src/cursor/abstract_cursor.ts

```diff
@@ -708,6 +773,12 @@ export abstract class AbstractCursor<
    * a significant refactor.
    */
   private async cursorInit(): Promise<void> {
+    if (this.cursorOptions.timeoutMS != null) {
+      this.timeoutContext = TimeoutContext.create({
+        serverSelectionTimeoutMS: this.client.options.serverSelectionTimeoutMS,
+        timeoutMS: this.cursorOptions.timeoutMS
+      });
+    }
```

Comment:
Re getting `serverSelectionTimeoutMS` from the client, the typescript doesn't give type hinting for that option, so users won't expect to be able to provide it.

Reply:
We don't always create a timeoutcontext because we construct a timeout context as needed in `executeOperation`. Also when running with legacy timeouts, there is no concept of a `timeoutMode`, so the driver wouldn't know when to refresh timeouts